src/tests/performance/connections.rs

```diff
@@ -55,168 +60,202 @@ async fn incoming_active_connections() {
+    // ├───┼─────┼──────────┼──────────┼────────────┼──────────┤
+    // │ 50│15000│        42│     14958│           0│      3.19│
+    // ├───┼─────┼──────────┼──────────┼────────────┼──────────┤
+    // │ 50│20000│        42│     19958│           0│      3.20│
+    // └───┴─────┴──────────┴──────────┴────────────┴──────────┘
+    //
+    // Zebra:
+    // ┌───┬─────┬──────────┬──────────┬────────────┬──────────┐
+    // │ N │  M  │ accepted │ rejected │ terminated │ time (s) │
+    // ├───┼─────┼──────────┼──────────┼────────────┼──────────┤
+    // │ 50│  100│       100│         0│           0│      0.11│
+    // ├───┼─────┼──────────┼──────────┼────────────┼──────────┤
+    // │ 50│ 1000│      1000│         0│           0│      0.66│
+    // ├───┼─────┼──────────┼──────────┼────────────┼──────────┤
+    // │ 50│ 5000│      3785│      1215│           0│      2.27│
+    // ├───┼─────┼──────────┼──────────┼────────────┼──────────┤
+    // │ 50│10000│      6146│      3854│           0│      7.58│
+    // └───┴─────┴──────────┴──────────┴────────────┴──────────┘
     //
-    // The example runs are pretty representative. In particular, zebra starts exhibiting weird
-    // behaviour at 15_000.
 
     /// maximum peers to configure node with
-    const MAX_PEERS: usize = 50;
-    /// number of test peer nodes to spin-up
-    const TEST_PEER_COUNT: usize = 100;
+    const MAX_PEERS: u16 = 50;
+
+    let peer_counts = vec![100u16, 1_000, 5_000, 10_000]; //15_000, 20_000];
 
-    // channel for peer event management (ensure buffer is more than large enough)
-    let (event_tx, mut event_rx) = tokio::sync::mpsc::channel::<PeerEvent>(TEST_PEER_COUNT * 3);
+    let mut all_stats = Vec::new();
 
     // start node
     let mut node: Node = Default::default();
     node.initial_action(Action::WaitForConnection)
-        .max_peers(MAX_PEERS)
+        .max_peers(MAX_PEERS as usize)
         .start()
         .await;
 
-    // start peer event manager
-    let event_manager = tokio::spawn(async move {
-        let mut stats = ConnectionStats::default();
-
-        loop {
-            match event_rx.recv().await.unwrap() {
-                PeerEvent::Rejected => stats.rejected += 1,
-                PeerEvent::Terminated => {
-                    stats.terminated += 1;
-                    // Since a connection cannot be terminated without first having connected,
-                    // there will always be a `last` item here.
-                    let prev_count = *stats.active_count.last().unwrap();
-                    stats.active_count.push(prev_count - 1);
+    for peers in peer_counts {
+        let test_start = tokio::time::Instant::now();
+        // channel for peer event management (ensure buffer is more than large enough)
+        let (event_tx, mut event_rx) = tokio::sync::mpsc::channel::<PeerEvent>(peers as usize * 3);
+
+        // start peer event manager
+        let event_manager = tokio::spawn(async move {
```

Comment:
We may not need this as the stats are protected by a mutex? This approach (without the mutex) may be slightly faster but I think simplicity is probably the best thing to strive for in tests?

Reply:
I either forgot to `metricify` or overwrote my changes here :/

Reply:
I must have missed this test case when updating for `metrics`